sources/expander/expander.c

```diff
@@ -37,8 +37,9 @@ static char	*expand_arg(char **env, char *arg, size_t i)
 		env_utils->key = ft_strjoin("$", env_utils->key);
 	if (arg && env_utils->value && is_arg_key(arg, env_utils->key))
 	{
-		free_env_values(env_utils->key, env_utils->value, env_utils);
-		return (arg = env_utils->value);
+		arg = env_utils->value;
```

Comment:
All values need to be passed to`free_env_values()` function.

Reply:
Issue is that freeing the values in this specific statement that the arg value would be lost, having no value.

[minishell]: echo $USER
� 
[minishell]: echo "$USER"
qtrinh 

you can try it out. not passing the env_utils->value to free_env_values retains the expanded env value and therefore, works correctly.